spectator-ext-aws/src/main/java/com/netflix/spectator/aws/SpectatorServiceMetricCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.aws;
+
+import com.amazonaws.metrics.ByteThroughputProvider;
+import com.amazonaws.metrics.ServiceLatencyProvider;
+import com.amazonaws.metrics.ServiceMetricCollector;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.impl.Preconditions;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A ServiceMetricCollector that captures service metrics in a Registry.
+ */
+public class SpectatorServiceMetricCollector extends ServiceMetricCollector {
+    private static final long NANOS_PER_MILLI = TimeUnit.MILLISECONDS.toNanos(1);
+    private final Registry registry;
+
+    /**
+     * Constructs a new SpectatorServiceMetricCollector.
+     *
+     * @param registry the Registry for this SpectatorServiceMetricCollector.
+     */
+    public SpectatorServiceMetricCollector(Registry registry) {
+        super();
+        this.registry = Preconditions.checkNotNull(registry, "registry");
+    }
+
+    @Override
+    public void collectByteThroughput(ByteThroughputProvider provider) {
+        if (provider.getByteCount() > 0) {
+            registry.counter(getThroughputBytesId(provider)).increment(provider.getByteCount());
+            registry.timer(getThroughputDurationId(provider)).record(provider.getDurationNano(), TimeUnit.NANOSECONDS);
+        }
+
+    }
+
+    @Override
+    public void collectLatency(ServiceLatencyProvider provider) {
+        final long durationNanos = toNanos(provider.getDurationMilli());
```

Comment:
Maybe add a comment that `provider.getDurationMilli()` returns a double and needs to be converted to nanos to avoid losing precision. When I first looked at this I was wondering why it didn't use use `TimeUnit.MILLISECONDS`.